### DISCUSSION TITLE
Prisma application on Cpanel hosted nodejs application

### DISCUSSION THREAD

**Discussion by saugatdai** [prisma/prisma#3163] [Q&A] [answered]
Most of the web hosting available are of the type shared. I wonder if prisma application can be deployed in cpanel hosting environment while developing Node.js application.

**Comment by ryands17** [comment, marked as answer]
Hey @saugatdai 👋 
Yes it can be. Deploying a Prisma application is no different than deploying a traditional Node app to Cpanel :)
You would just need to add the specific binary target as deined [here](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/generators#binary-targets) based on the operating system.

**Reply by torver213** [reply]
Is there any example or article that points to this? Where can get a example on how to deploy a Nodejs Prisma mysql database on cpanel?

**Reply by anth0ni3** [reply]
Did you see any?

**Reply by torver213** [reply]
> Did you see any?

Not yet.. still working on that...

**Reply by hasanshahriar32** [reply]
hi, did you guys found any example/ tutorial to this?
how to deploy a Nodejs Prisma mysql database on cpanel

**Reply by ghost** [reply]
Can we have any update on this ?

**Comment by walidboulanouar** [comment]
hello i have a problem when i want to run nestjs app with prisma, when i run npx prisma db push command it shows me error:
RangeError: WebAssembly.Instance(): Out of memory: wasm memory

**Reply by janpio** [reply]
Is that related to Cpanel in any way?
Where and how are you running `db push`?

**Reply by walidboulanouar** [reply]
i want to deploy the app  in cpanel : so i can run command  on node js setup, i put this command in package.json, the it return this:
returncode: 1
stdout:
> sedi_hisham_back_end@0.0.1 prisma
> npx prisma db push
stderr:

**Reply by walidboulanouar** [reply]
I am really want to find solution how i can deploy prisma on cpanel, because i really want to generate Prisma client files in node modules

**Reply by walidboulanouar** [reply]
the process of dev is so good but the problem is that i need to run it on cpanel

**Reply by walidboulanouar** [reply]
you can see here where i run 
![image](https://user-images.githubusercontent.com/72222865/207623300-008f7f44-c86b-402a-88e9-8386cb5904d1.png)

**Reply by walidboulanouar** [reply]
![image](https://user-images.githubusercontent.com/72222865/207623373-80b32a55-45d9-47ef-968b-bda2ed1a0cce.png)

**Reply by janpio** [reply]
Please open a bug report about this. We have never seen this error message, so we need all the information you can provide when opening the bug report issue.

**Reply by walidboulanouar** [reply]
Yeah i opened a bug report, could you tell me the steps how i can run nestjs +prisma app on cpanel, i may forget a mendatory step, but prisma with nestjs are running properly on my computer, but i really need to deploy it

**Reply by walidboulanouar** [reply]
![image](https://user-images.githubusercontent.com/72222865/207646987-c8996de8-aebb-4023-a9a3-eecef33e3898.png)

**Reply by janpio** [reply]
What you show looks perfectly fine, it seems your build step just does not have enough memory to actually run Prisma.

**Reply by coderboysobuj** [reply]
> ![image](https://user-images.githubusercontent.com/72222865/207646987-c8996de8-aebb-4023-a9a3-eecef33e3898.png)

I Have same problem with Express Typescript and prisma

**Reply by walidboulanouar** [reply]
yeaah the same brother, just the command related to prisma that dosn't work on cpanel

**Reply by JohnMwanyika** [reply]
anyone who was able to come up up with a solution on this ..really frustrating compared to hosting on other platforms like Render works perfectly

**Reply by JohnMwanyika** [reply]
> What you show looks perfectly fine, it seems your build step just does not have enough memory to actually run Prisma.

I doubt,,this is not related to memory,,I am deploying mine on a VPS with very high specs but am experiencing the same issue

**Reply by Francoois** [reply]
Have you reached a limit on the number of processes in CPanel (NPROC in the metrics) ? Have you checked your pnpm store config, or tried using npm instead ?

**Reply by JohnMwanyika** [reply]
> Have you reached a limit on the number of processes in CPanel (NPROC in the metrics) ? Have you checked your pnpm store config, or tried using npm instead ?

Number of processes is 0/100 (0%) what do you mean using npm,,am trying to execute "prisma generate"

**Reply by Francoois** [reply]
My bad! I said that because on the screenshot the installed prisma is located in pnpm folder, and in the past I've had issues using prisma with pnpm on cloud linux / cpanel. But the screenshot wasn't yours so no reason to think there is a connection

**Reply by anan474** [reply]
Found the solution for this, the problem is on the cpanel and/or virtual terminal provided in cpanel. All you need to do is to run the command (prisma generate, etc) from ssh client (putty, termius, etc).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
